jung-samples/src/main/java/edu/uci/ics/jung/samples/TreeCollapseDemo.java

```diff
@@ -14,20 +14,19 @@
 import edu.uci.ics.jung.graph.TreeNetworkBuilder;
 import edu.uci.ics.jung.layout.algorithms.BalloonLayoutAlgorithm;
 import edu.uci.ics.jung.layout.algorithms.RadialTreeLayoutAlgorithm;
-import edu.uci.ics.jung.layout.algorithms.TreeLayoutAlgorithm;
+import edu.uci.ics.jung.layout.algorithms.immutable.TreeLayoutAlgorithm;
```

Comment:
I'm thinking of changing all of the layout algorithms to use an immutable Point class and tossing out the PointModel work plus all of the 3rd dimension (z and depth) support in LayoutModel.

Reply:
OK, understood, thanks.  Can you say more about what's caused you to ditch the PointModel work?  It seemed to me like a reasonable solution to the interoperation problem.

Reply:
I started making a BarnesHutQuadTree, which requires a Point and a Rectangle. During refactoring, I have strived to use x, y instead of Point where possible, especially trying to remove places where I create a Point only to later pull out the x, y.
That led me to try out a Point class. It seems to have resulted in cleaner code, and the only place I heed Point2D is over in visualization where I am using AffineTransforms.